src/develop/develop.c

```diff
@@ -1470,6 +1470,8 @@ void dt_dev_read_history_ext(dt_develop_t *dev, const int imgid, gboolean no_ima
 
   dev->iop_order_version = 0;
 
+  dt_ioppr_convert_onthefly(imgid);
```

Comment:
Maybe this routine could return the iop_order_version for imgid and then we can remove the code below that is reading back iop_order_version from db?

Reply:
Absolutely, i had it in mind but didn't do so until i know we are safe in the on-the-fly stuff and don't have hard ends.

Reply:
dt_ioppr_convert_onthefly now returns the iop_order_version from db ...

Reply:
So let's use the return value here now. And remove code below.